tests/test_estimators.py

```diff
@@ -118,3 +120,33 @@ def test_estimator_output_type_consistency(output_mode, x_type, estimator):
         ref_type = type(ref_estimator.fit(X, y).predict(X))
 
     assert sknnr_type is ref_type  # noqa: E721
+
+
+@pytest.mark.parametrize("estimator", TEST_YFIT_ESTIMATORS)
+def test_yfit_is_stored(estimator):
+    """Test that y_fit is stored when passed."""
+    X, y = load_moscow_stjoes(return_X_y=True)
+    # Arbitrary split with a fixed constant to prevent zero sum rows
+    y_fit = y[:, 10:] + 0.1
+    y = y[:, :10] + 0.1
+
+    est = estimator().fit(X, y)
+    assert est.y_fit_ is None
+    est.fit(X, y, y_fit=y_fit)
+    assert_array_equal(est.y_fit_, y_fit)
+
+
+@pytest.mark.parametrize("estimator", TEST_YFIT_ESTIMATORS)
+def test_yfit_affects_prediction(estimator):
+    """Test that y_fit affects predictions when passed."""
+    X, y = load_moscow_stjoes(return_X_y=True)
+    # Arbitrary split with a fixed constant to prevent zero sum rows
+    y_fit = y[:, 10:] + 0.1
+    y = y[:, :10] + 0.1
```

Comment:
I don't love the duplication, but also didn't want to test multiple things at once. Let me know if you think there's a better approach.

Reply:
I suppose you could put the data into a fixture and use that, e.g.
```
@pytest.fixture
def fit_data():
    X, y = load_moscow_stjoes(return_X_y=True)
    # Arbitrary split with a fixed constant to prevent zero sum rows
    y_fit = y[:, 10:] + 0.1
    y = y[:, :10] + 0.1
    return X, y, y_fit

...

@pytest.mark.parametrize("estimator", TEST_YFIT_ESTIMATORS)
def test_yfit_is_stored(estimator, fit_data):
    """Test that y_fit is stored when passed."""
    X, y, y_fit = fit_data
    ...

@pytest.mark.parametrize("estimator", TEST_YFIT_ESTIMATORS)
def test_yfit_affects_prediction(estimator, fit_data):
    """Test that y_fit affects predictions when passed."""
    X, y, y_fit = fit_data
   ...
```
but not sure that helps the readibility.  I'm fine with the duplication.

Reply:
I think a fixture is a good idea!